OSDOCS#6319: ARM support for GCP

kelbrown20 opened the pull request:
**For Version:** 4.14+
**Issue:** [OSDOCS-6319](https://issues.redhat.com//browse/OSDOCS-6319)

**Previews:**  

[Selecting a cluster installation method and preparing it for users -> Supported installation methods for different platforms ](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing-preparing.html)

Installing 
- [Installing a cluster on GCP with customizations -> Tested instance types for GCP on 64-bit ARM infrastructures](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations.html)
- [Installing a cluster on GCP with Network customizations -> Tested instance types for GCP on 64-bit ARM infrastructures](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-network-customizations.html#installation-gcp-tested-machine-types-arm_installing-gcp-network-customizations)
- [Installing a cluster on GCP with Network customizations -> Enabling Shielded VMs](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-network-customizations#installation-gcp-enabling-shielded-vms_installing-gcp-network-customizations)
- [Installing a cluster on GCP in a restricted network -> Tested instance types for GCP on 64-bit ARM infrastructures ](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.html#installation-gcp-tested-machine-types-arm_installing-restricted-networks-gcp-installer-provisioned)
- [Installing a cluster on GCP into an existing VPC -> Tested instance types for GCP on 64-bit ARM infrastructures ](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-vpc.html#installation-gcp-tested-machine-types-arm_installing-gcp-vpc)
- [Installing a private cluster on GCP -> Tested instance types for GCP in 64-bit ARM infrastructures ](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-private.html#installation-gcp-tested-machine-types-arm_installing-gcp-private)

[Installing on GCP -> Installation configuration parameters for GCP](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installation-config-parameters-gcp#installation-configuration-parameters-optional_installation-config-parameters-gcp)
Updates made in these parameter sections in _Optional Parameters_

-  _Optional Parameters_
            - `compute.architecture`
            - `controlPlane.architecture`
-  _Additional Google Cloud Platform (GCP) configuration parameters_
            - `platform.gcp.defaultMachinePlatform.zones`
            - `controlPlane.platform.gcp.zones`
            - `compute.platform.gcp.zones`


 
**QE review:** 
- [x] QE approval

r4f4 commented:
@kelbrown20 when I click on `Machine series for 64-bit ARM machines` it just shows me an empty list. Is that expected?

r4f4 commented:
Oh, it's because the Installer PR hasn't merged yet, right? Apart from that
/lgtm

kelbrown20 commented:
@aleskandro @lwan-wanglin Hello! Can you PTAL at this PR from the GCP and ARM QE side?

kelbrown20 commented:
@jeffdyoung Hey Jeff! can you also PTAL at this PR when you get the chance?

lwan-wanglin commented on installing/installing-preparing.adoc:
```diff
@@ -447,7 +452,7 @@ endif::openshift-origin[]
 ifndef::openshift-origin[]
 |===
-||Alibaba |AWS (x86_64) |AWS (arm64) |Azure (x86_64) |Azure (arm64) |Azure Stack Hub |GCP |Nutanix |{rh-openstack} |Bare metal (x86_64) |Bare metal (arm64) |vSphere |VMC |IBM Cloud VPC |{ibmzProductName} |{ibmzProductName} with {op-system-base} KVM |{ibmpowerProductName} |Platform agnostic
+||Alibaba |AWS (64-bit x86) |AWS (64-bit ARM) |Azure (64-bit x86) |Azure (64-bit ARM) |Azure Stack Hub |GCP |Nutanix |{rh-openstack} |Bare metal (64-bit x86) |Bare metal (64-bit ARM) |vSphere |IBM Cloud VPC |{ibmzProductName} |{ibmzProductName} with {op-system-base} KVM |{ibmpowerProductName} |Platform agnostic
```
@kelbrown20 we also support arm upi install on GCP, so we can updage `GCP` to `GCP (64-bit x86) |GCP (64-bit ARM)`

lwan-wanglin commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -639,11 +639,11 @@ ifndef::aws,bare,ibm-z,ibm-power,azure,ibm-power-vs[]
 
-ifdef::aws,bare,azure[]
+ifdef::aws,bare,azure,gcp[]
 |`controlPlane.architecture`
```
it seems double `controlPlane.architecture` field in the doc [installation-config-parameters-gcp](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installation-config-parameters-gcp#installation-configuration-parameters-optional_installation-config-parameters-gcp)

lwan-wanglin commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -575,11 +575,11 @@ ifndef::aws,bare,ibm-power,ibm-z,azure,ibm-power-vs[]
 
-ifdef::aws,bare,azure[]
+ifdef::aws,bare,azure,gcp[]
 |`compute.architecture`
```
it seems double `compute.architecture` field in the doc [installation-config-parameters-gcp](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installation-config-parameters-gcp#installation-configuration-parameters-optional_installation-config-parameters-gcp)

lwan-wanglin commented:
@kelbrown20 we now have a bug [OCPBUGS-17904](https://issues.redhat.com/browse/OCPBUGS-17904) for  [enabling shielded VMs](https://62253--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-network-customizations.html#installation-gcp-enabling-shielded-vms_installing-gcp-network-customizations) arm on gcp, it's a kernel bug, but it influence we install a shielded arm cluster on GCP, should we document it somewhere?

kelbrown20 commented:
Thank you for the review! I just fixed and updated the comments you mentioned 
Regarding: 
> enabling shielded VMs

Is there a different process to enable shielded VMs on ARM? Or is it completely unsupported on ARM?

lwan-wanglin commented:
> 
> Is there a different process to enable shielded VMs on ARM? Or is it completely unsupported on ARM?

it's completely unsupported on ARM

lwan-wanglin commented on installing/installing-preparing.adoc:
```diff
@@ -497,6 +504,7 @@ ifndef::openshift-origin[]
 |
 |
 |xref:../installing/installing_gcp/installing-restricted-networks-gcp.adoc#installing-restricted-networks-gcp[&#10003;]
+|xref:../installing/installing_gcp/installing-restricted-networks-gcp.adoc#installing-restricted-networks-gcp[&#10003;]
```
we support below `Shared VPC hosted outside of cluster project` on ARM too

lwan-wanglin commented:
others look good to me! thanks @kelbrown20

kelbrown20 commented on modules/installation-gcp-enabling-shielded-vms.adoc:
```diff
@@ -14,6 +14,11 @@
 = Enabling Shielded VMs
 You can use Shielded VMs when installing your cluster. Shielded VMs have extra security features including secure boot, firmware and integrity monitoring, and rootkit detection. For more information, see Google's documentation on link:https://cloud.google.com/shielded-vm[Shielded VMs].
 
+[NOTE]
```
@lwan-wanglin I just added this note for the Shielded VM section, WDYT?

lwan-wanglin commented:
/lgtm

ShaunaDiaz commented:
/label peer-review-in-progress

ShaunaDiaz commented:
/remove-label peer-review-needed

ShaunaDiaz commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -577,17 +577,17 @@ accounts for the dramatically decreased machine performance.
-ifdef::aws,bare,azure[]
+ifdef::aws,bare,azure,gcp[]
 |`controlPlane.architecture`
 |Determines the instruction set architecture of the machines in the pool. Currently, clusters with varied architectures are not supported. All pools must specify the same architecture. Valid values are `amd64` and `arm64`. See _Supported installation methods for different platforms_ in _Installing_ documentation for information about instance availability.
```
Out of scope, really, but maybe you could replace the sentence with an xref to https://docs.openshift.com/container-platform/4.13/installing/installing-preparing.html#supported-installation-methods-for-different-platforms?

kelbrown20 replied:
I would want to but Ik we can't pus xrefs in modules :'(

ShaunaDiaz commented:
One suggestion, looks great to me!

ShaunaDiaz commented:
/label peer-review-done

ShaunaDiaz commented:
/remove-label peer-review-in-progress

kelbrown20 commented:
/cherrypick enterprise-4.14
